lib/Dialect/Ibis/IbisOps.cpp

```diff
@@ -20,71 +20,46 @@ using namespace circt;
 using namespace ibis;
 
 template <typename TSymAttr>
-ParseResult parseClassRefFromName(OpAsmParser &parser, Type &classRefType,
+ParseResult parseScopeRefFromName(OpAsmParser &parser, Type &ScopeRefType,
                                   TSymAttr sym) {
   // Nothing to parse, since this is already encoded in the child symbol.
-  classRefType = ClassRefType::get(parser.getContext(), sym);
+  ScopeRefType = ScopeRefType::get(parser.getContext(), sym);
   return success();
 }
 
 template <typename TSymAttr>
-void printClassRefFromName(OpAsmPrinter &p, Operation *op, Type type,
+void printScopeRefFromName(OpAsmPrinter &p, Operation *op, Type type,
                            TSymAttr sym) {
   // Nothing to print since this information is already encoded in the child
   // symbol.
 }
 
 //===----------------------------------------------------------------------===//
-// ClassOp
+// ScopeOpInterface
 //===----------------------------------------------------------------------===//
 
-ParseResult ClassOp::parse(OpAsmParser &parser, OperationState &result) {
-  // Parse signature
-  StringAttr className;
-  if (parser.parseSymbolName(className, mlir::SymbolTable::getSymbolAttrName(),
-                             result.attributes))
-    return failure();
+FailureOr<Value> circt::ibis::detail::getThisFromScope(Operation *op) {
+  auto scopeOp = cast<ScopeOpInterface>(op);
+  auto thisOps = scopeOp.getBodyBlock()->getOps<ibis::ThisOp>();
+  if (thisOps.empty())
+    return op->emitOpError("must contain a 'ibis.this' operation");
 
-  // parse "this"
-  OpAsmParser::Argument thisArg;
-  if (parser.parseLParen() ||
-      parser.parseArgument(thisArg, /*allowType=*/false) ||
-      parser.parseRParen())
-    return failure();
+  if (std::next(thisOps.begin()) != thisOps.end())
```

Comment:
Why not use `thisOps.size()` rather than comparing iterators?

Reply:
`getOps` returns an `iterator_range` which has `begin, end, empty`. So one can either use `std::distance` or the above... went with the above, although `std::distance` is probably the better, more readable, option to infer the size.